share/p2p/discovery/options.go

```diff
@@ -14,45 +21,56 @@ type Parameters struct {
 	// Set -1 to disable.
 	// NOTE: only full and bridge can advertise themselves.
 	AdvertiseInterval time.Duration
+	// Tag is used as rondezvous point for discovery service
+	Tag string
+}
+
+// options is the set of options that can be configured for the Discovery module
+type options struct {
+	// onUpdatedPeers will be called on peer set changes
+	onUpdatedPeers OnUpdatedPeers
 }
 
 // Option is a function that configures Discovery Parameters
-type Option func(*Parameters)
+type Option func(*options)
 
 // DefaultParameters returns the default Parameters' configuration values
 // for the Discovery module
-func DefaultParameters() Parameters {
-	return Parameters{
+func DefaultParameters() *Parameters {
+	return &Parameters{
 		PeersLimit:        5,
 		AdvertiseInterval: time.Hour,
+		//TODO: remove fullNodesTag default value once multiple tags are supported
+		Tag: fullNodesTag,
```

Comment:
[blocking]
This will break our configs and will be removed anyway. Let's simply set `fullNodesTag` if the Tag is empty with the same todo that this should change once we introduce support for multiple

Reply:
Sounds good. I've moved tag to constructor, so it is not optional anymore and have to be set to non-zero value. So there is no need for this TODO anymore.  
Also cleaned up a bit params validation, because it looked bad that some params values were used to only configure tests. For example PeerLimit == 0 used to disable discovery scanning, when it was possible to just skip call `Start`.  This will achieve same result without weird params usage. Same for AdvertiseInterval == -1, when it was possible to just skip `Advertise` call to disable advertising.